server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteRestoreSnapshotIT.java

```diff
@@ -327,16 +329,22 @@ public void testRemoteStoreCustomDataOnIndexCreationAndRestore() {
         validatePathType(indexName2, PathType.HASHED_PREFIX, PathHashAlgorithm.FNV_1A);
 
         // Validating that custom data has not changed for indexes which were created before the cluster setting got updated
-        validatePathType(indexName1, PathType.FIXED, PathHashAlgorithm.FNV_1A);
+        validatePathType(indexName1, PathType.FIXED);
```

Comment:
Can we add one test to restore on the same index, basically `create idx -> take snapshot -> update cluster settings -> close idx -> restore idx from snapshot -> validate that path type is still PathType.FIXED`.

Reply:
`testRemoteStoreCustomDataOnIndexCreationAndRestore`already covers this case. I will adding more ITs as part of https://github.com/opensearch-project/OpenSearch/issues/12766 after https://github.com/opensearch-project/OpenSearch/issues/13074 (All changes are wired up).

Reply:
no, we are using different index there during restore. we need to close the same index and restore it from snapshot. verified that it will work fine from code perspective, but it would be good to have a test, maybe in the above followup PRs.

Reply:
Ohk, we are restoring but as a new index. I get your point, will try to put this in the same PR.